Test failures with Python 3.13.0b1 in 4.4.0

### Things to check first

- [X] I have searched the existing issues and didn't find my bug already reported there

- [X] I have checked that my bug is still present in the latest release


### AnyIO version

4.4.0

### Python version

3.13.0b1

### What happened?

```
______________________________________________________________________________________ TestPath.test_properties[asyncio] _______________________________________________________________________________________
tests/test_fileio.py:103: in test_properties
    assert not missing
E   AssertionError: assert not {'from_uri', 'full_match', 'parser'}
___________________________________________________________________________________ TestPath.test_properties[asyncio+uvloop] ___________________________________________________________________________________
tests/test_fileio.py:103: in test_properties
    assert not missing
E   AssertionError: assert not {'from_uri', 'full_match', 'parser'}
________________________________________________________________________________________ TestPath.test_properties[trio] ________________________________________________________________________________________
tests/test_fileio.py:103: in test_properties
    assert not missing
E   AssertionError: assert not {'from_uri', 'full_match', 'parser'}
__________________________________________________________________________________________ TestPath.test_is_reserved ___________________________________________________________________________________________
tests/test_fileio.py:274: in test_is_reserved
    assert Path("nul").is_reserved() == expected_result
src/anyio/_core/_fileio.py:445: in is_reserved
    return self._path.is_reserved()  
/usr/lib64/python3.13/pathlib/_local.py:423: in is_reserved
    warnings.warn(msg, DeprecationWarning, stacklevel=2)
E   DeprecationWarning: pathlib.PurePath.is_reserved() is deprecated and scheduled for removal in Python 3.15. Use os.path.isreserved() to detect reserved paths on Windows.
```

I expected these four tests to pass, as they do on Python 3.12.

### How can we reproduce the bug?

```
$ gh repo clone agroholm/anyio
$ cd anyio
$ python3.13 --version
Python 3.13.0b1
$ python3.13 -m venv _e
$ . _e/bin/activate
```

Before proceeding further, we need to build `uvloop` ourselves, since the released sdist doesn’t build on Python 3.13 and there are no binary wheels on PyPI for Python 3.13 yet. Furthermore, we need to patch it for Cython 3 as proposed in https://github.com/MagicStack/uvloop/pull/587. This reflects [how `uvloop` is packaged in Fedora](https://src.fedoraproject.org/rpms/python-uvloop/blob/3a3cfe152992e111eac8e79a7fb7ea8ead71beb9/f/587.patch).

```
(_e) $ cd ..
(_e) $ gh repo clone MagicStack/uvloop
(_e) $ cd uvloop
(_e) $ git submodule update --init --recursive
(_e) $ git remote add alan-brooks https://github.com/alan-brooks/uvloop.git
(_e) $ git fetch --all
(_e) $ git branch cython3
(_e) $ git checkout cython3
(_e) $ git merge alan-brooks/cython3_update
(_e) $ sed -r -i 's/(Cython\(>=.*,<)(.*)(\))/\14\3/' pyproject.toml
(_e) $ git add pyproject.toml
(_e) $ git commit -m 'Allow Cython 3'
(_e) $ pip install .
```

Furthermore, to avoid errors about `undefined symbol: _PyErr_WriteUnraisableMsg`, we need `cffi` to contain https://github.com/python-cffi/cffi/commit/49127c6929bfc7186fbfd3819dd5e058ad888de4. The easiest way is to manually install 1.17.0rc1.

```
(_e) $ pip install cffi==1.17.0rc1
```

 Now we can continue:

```
(_e) $ cd ../anyio
(_e) $ pip install -e .[test]
(_e) $ python -m pytest
```

```
=========================================================================================== short test summary info ============================================================================================
SKIPPED [1] tests/test_debugging.py:102: Generator based coroutines have been removed in Python 3.11
SKIPPED [1] tests/test_fileio.py:135: Drive only makes sense on Windows
SKIPPED [1] tests/test_fileio.py:177: Only makes sense on Windows
SKIPPED [3] tests/test_fileio.py:366: os.lchmod() is not available
SKIPPED [1] tests/test_taskgroups.py:964: Generator based coroutines have been removed in Python 3.11
SKIPPED [1] tests/test_taskgroups.py:985: Generator based coroutines have been removed in Python 3.11
XFAIL tests/test_from_thread.py::TestBlockingPortal::test_from_async[trio-trio] - reason: known bug (#525)
XFAIL tests/test_subprocesses.py::test_process_aexit_cancellation_closes_standard_streams[asyncio] - #671 needs to be resolved first
XFAIL tests/test_subprocesses.py::test_process_aexit_cancellation_closes_standard_streams[asyncio+uvloop] - #671 needs to be resolved first
XFAIL tests/test_taskgroups.py::test_fail_after_scope_cancelled_before_timeout[asyncio] - There is currently no way to tell if cancellation happened due to timeout explicitly if the deadline has been exceeded
XFAIL tests/test_taskgroups.py::test_fail_after_scope_cancelled_before_timeout[asyncio+uvloop] - There is currently no way to tell if cancellation happened due to timeout explicitly if the deadline has been exceeded
XFAIL tests/test_taskgroups.py::test_fail_after_scope_cancelled_before_timeout[trio] - There is currently no way to tell if cancellation happened due to timeout explicitly if the deadline has been exceeded
============================================================================ 55 failed, 1379 passed, 8 skipped, 6 xfailed in 36.48s ============================================================================
```

I’m not as worried about the failures in `tests/test_sockets.py` due to `socket.gaierror: [Errno -2] Name or service not known` – we run tests offline in Fedora, so we don’t see these, and these reproduce in Python 3.12 – but failures in `tests/test_fileio.py` appear to be real regressions in Python 3.13.

--- a/docs/versionhistory.rst
+++ b/docs/versionhistory.rst
@@ -5,8 +5,8 @@ This library adheres to `Semantic Versioning 2.0 <http://semver.org/>`_.
 
 **UNRELEASED**
 
-- Added support for the the ``from_uri()``, ``full_match()``, ``parser``
-  methods/properties in ``anyio.Path``, newly added in Python 3.13
+- Added support for the ``from_uri()``, ``full_match()``, ``parser`` methods/properties
+  in ``anyio.Path``, newly added in Python 3.13
 
 **4.4.0**
 

--- a/docs/versionhistory.rst
+++ b/docs/versionhistory.rst
@@ -3,6 +3,11 @@ Version history
 
 This library adheres to `Semantic Versioning 2.0 <http://semver.org/>`_.
 
+**UNRELEASED**
+
+- Added support for the the ``from_uri()``, ``full_match()``, ``parser``
+  methods/properties in ``anyio.Path``, newly added in Python 3.13
+
 **4.4.0**
 
 - Added the ``BlockingPortalProvider`` class to aid with constructing synchronous

--- a/pyproject.toml
+++ b/pyproject.toml
@@ -51,7 +51,7 @@ test = [
     "pytest >= 7.0",
     "pytest-mock >= 3.6.1",
     "trustme",
-    "uvloop >= 0.17; platform_python_implementation == 'CPython' and platform_system != 'Windows' and python_version < '3.13'",
+    "uvloop >= 0.17; platform_python_implementation == 'CPython' and platform_system != 'Windows'",
 ]
 doc = [
     "packaging",

--- a/pyproject.toml
+++ b/pyproject.toml
@@ -51,7 +51,7 @@ test = [
     "pytest >= 7.0",
     "pytest-mock >= 3.6.1",
     "trustme",
-    "uvloop >= 0.17; platform_python_implementation == 'CPython' and platform_system != 'Windows'",
+    "uvloop >= 0.17; platform_python_implementation == 'CPython' and platform_system != 'Windows' and python_version < '3.13'",
 ]
 doc = [
     "packaging",
@@ -87,7 +87,7 @@ extend-select = [
 "required-imports" = ["from __future__ import annotations"]
 
 [tool.mypy]
-python_version = "3.12"
+python_version = "3.13"
 strict = true
 ignore_missing_imports = true
 disallow_any_generics = false

--- a/src/anyio/_core/_fileio.py
+++ b/src/anyio/_core/_fileio.py
@@ -358,8 +358,28 @@ class Path:
     def as_uri(self) -> str:
         return self._path.as_uri()
 
-    def match(self, path_pattern: str) -> bool:
-        return self._path.match(path_pattern)
+    if sys.version_info >= (3, 13):
+        parser = pathlib.Path.parser  # type: ignore[attr-defined]
+
+        @classmethod
+        def from_uri(cls, uri: str) -> Path:
+            return Path(pathlib.Path.from_uri(uri))  # type: ignore[attr-defined]
+
+        def full_match(
+            self, path_pattern: str, *, case_sensitive: bool | None = None
+        ) -> bool:
+            return self._path.full_match(  # type: ignore[attr-defined]
+                path_pattern, case_sensitive=case_sensitive
+            )
+
+        def match(
+            self, path_pattern: str, *, case_sensitive: bool | None = None
+        ) -> bool:
+            return self._path.match(path_pattern, case_sensitive=case_sensitive)
+    else:
+
+        def match(self, path_pattern: str) -> bool:
+            return self._path.match(path_pattern)
 
     def is_relative_to(self, other: str | PathLike[str]) -> bool:
         try:

--- a/tests/test_fileio.py
+++ b/tests/test_fileio.py
@@ -186,6 +186,15 @@ class TestPath:
         else:
             assert Path("/foo/bar").as_uri() == "file:///foo/bar"
 
+    @pytest.mark.skipif(
+        sys.version_info < (3, 13),
+        reason="Path.from_uri() is only available on Python 3.13+",
+    )
+    def test_from_uri(self) -> None:
+        path = Path.from_uri("file:///foo/bar")
+        assert isinstance(path, Path)
+        assert path.as_uri() == "file:///foo/bar"
+
     async def test_cwd(self) -> None:
         result = await Path.cwd()
         assert isinstance(result, Path)
@@ -269,6 +278,7 @@ class TestPath:
         assert not await Path("/gfobj4ewiotj").is_mount()
         assert await Path("/").is_mount()
 
+    @pytest.mark.filterwarnings("ignore::DeprecationWarning")
     def test_is_reserved(self) -> None:
         expected_result = platform.system() == "Windows"
         assert Path("nul").is_reserved() == expected_result
@@ -338,6 +348,14 @@ class TestPath:
     def test_joinpath(self) -> None:
         path = Path("/foo").joinpath("bar")
         assert path == Path("/foo/bar")
+
+    @pytest.mark.skipif(
+        sys.version_info < (3, 13),
+        reason="Path.full_match() is only available on Python 3.13+",
+    )
+    def test_fullmatch(self) -> None:
+        assert Path("/foo/bar").full_match("/foo/*")
+        assert not Path("/foo/bar").full_match("/baz/*")
 
     def test_match(self) -> None:
         assert Path("/foo/bar").match("/foo/*")

